framework/include/base/Assembly.h

```diff
@@ -1394,6 +1416,17 @@ class Assembly
     return diag;
   }
 
+  /**
+   * Returns the current elem/volume quadrature rule.  The current subdomain
+   * (as set via setCurrentSubdomainID is used to determine the correct rule.
+   */
+  inline QBase * qruleElem(unsigned int dim) { return qrules(dim).vol.get(); }
```

Comment:
If this is going to be public, then I’d like to see this return a const QBase * and have the method marked const.  Then if you want to be able to use this for setting purposes within Assembly, then you could add a private non-const method overload that returns a QBase *. Same for the face method

Reply:
I actually tried to make them const at first.  But after attaching a quadrature rule to an fe object - they get init'd - so any actually used quadrature rules (which is the purpose of these accessors) will need to be available as non-const.

Reply:
What about about adding in `Assembly`
```
void attachQRuleElem(FEBase & fe, unsigned int dim)
{
  fe->attach_quadrature_rule(qRuleElem(dim));
}
```
I just think that this non-const accessor could really be abused by users. Someone could just willy nilly set the qrule to something crazy.

Reply:
Or sorry, they couldn't change the Assembly qrule to a different object, but they could mess with the qrule through the `get_points,get_weights,etc.` APIs. It's probably unlikely but I get paranoid

Reply:
Yeah.  Too bad QBase doesn't have a virtual copy-constructor pattern clone func we can call to just copy it.  That would be the best solution I think.

Reply:
Okay - this should be addressed.